core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -57,12 +57,13 @@
 
 /** A communication channel for making outgoing RPCs. */
 @ThreadSafe
-public final class ChannelImpl extends Channel {
+public final class ChannelImpl implements Channel {
   private static final Logger log = Logger.getLogger(ChannelImpl.class.getName());
   private final ClientTransportFactory transportFactory;
   private final ExecutorService executor;
   private final String userAgent;
   private final Object lock = new Object();
+  private final ClientCallFactory callFactory = new InternalCallFactory();
```

Comment:
Given this is the only instance of InternalCallFactory, I slightly prefer it being an anonymous class.